internal/coreinternal/scraperinttest/scraperint.go

```diff
@@ -59,49 +59,21 @@ type IntegrationTest struct {
 }
 
 func (it *IntegrationTest) Run(t *testing.T) {
-	ctx := context.Background()
-
-	var ci *ContainerInfo
-	var ciErr error
-	if it.containerRequest != nil {
-		require.NoError(t, it.containerRequest.Validate())
-		for _, port := range it.containerRequest.ExposedPorts {
-			require.False(t, strings.ContainsRune(port, ':'), errExposedPort)
-		}
+	it.validate(t)
 
-		var container testcontainers.Container
-		var containerErr error
-		defer func() {
-			if t.Failed() && containerErr != nil {
-				t.Error(containerErr.Error())
-			}
-		}()
-		require.Eventually(t, func() bool {
-			container, containerErr = testcontainers.GenericContainer(ctx,
-				testcontainers.GenericContainerRequest{
-					ContainerRequest: *it.containerRequest,
-					Started:          true,
-				})
-			return containerErr == nil
-		}, it.createContainerTimeout, time.Second)
-		defer func() {
-			require.NoError(t, container.Terminate(context.Background()))
-		}()
-
-		ci, ciErr = containerInfo(ctx, container, it.containerRequest.ExposedPorts)
-		require.NoError(t, ciErr)
-	}
+	ci := it.createContainers(t)
+	defer ci.terminate(t)
 
 	cfg := it.factory.CreateDefaultConfig()
 	it.customConfig(t, cfg, ci)
 	sink := new(consumertest.MetricsSink)
 	settings := receivertest.NewNopCreateSettings()
 
-	rcvr, err := it.factory.CreateMetricsReceiver(ctx, settings, cfg, sink)
+	rcvr, err := it.factory.CreateMetricsReceiver(context.Background(), settings, cfg, sink)
```

Comment:
Is it important now to use a new context for each of these functions instead of the same context?

Reply:
I'm not certain, but I don't think it's terribly important in these tests. I played around with contexts a bit here but it felt very awkward to pass around both `testing.T` and `context.Context`. That said, I'd like to do it the right way, but I figured it could be worth revisiting later.